### PR TITLE
[fastcgs] Huge page versions

### PR DESCRIPTION
**Pull request by alifahmed**
Enables huge page on fastcgs. Will keep normal or this version based on the performance difference. Depends on PR #247.

### CODE REVIEW

File: .github/workflows/fuzzers.yml [hunk 1]
```diff
@@ -36,7 +36,9 @@ jobs:
           - entropic
           - fairfuzz
           - fastcgs
+          - fastcgs_huge
```
Comment (jonathanmetzman) [inline .github/workflows/fuzzers.yml:39]: Do we need to test this feature with both? Or do you think one is enough to tell whether it is useful?

**Comment by jonathanmetzman** [comment]
So is this PR downloading a new version of fastcgs that can use these large pages?

File: .github/workflows/fuzzers.yml (hunk 1 above)
Comment (alifahmed) [reply .github/workflows/fuzzers.yml:39]: One is actually enough to test this feature. I wanted to compare huge page versions of fastcgs and fastcgs_lm, that's why added both. Please let me know if I should remove one of them.

**Comment by alifahmed** [comment]
> 
> 
> So is this PR downloading a new version of fastcgs that can use these large pages?

Yes, the current fastcgs/fastcgs_lm does not have large pages enabled. This PR adds fastcgs_huge/fastcgs_lm_huge with large pages enabled, everything else being same.

**Comment by alifahmed** [comment]
@lszekeres @jonathanmetzman I rerun the CI, but shmget() still failing to access huge pages. I am not sure where the problem is. Any idea what went wrong?

**Comment by jonathanmetzman** [comment]
> @lszekeres @jonathanmetzman I rerun the CI, but shmget() still failing to access huge pages. I am not sure where the problem is. Any idea what went wrong?

I personally wouldn't know the answer to this, but maybe the CI environment is just weird and this won't happen in production? What happens if you run it locally? Does it work then?

**Comment by lszekeres** [comment]
> > @lszekeres @jonathanmetzman I rerun the CI, but shmget() still failing to access huge pages. I am not sure where the problem is. Any idea what went wrong?
> 
> I personally wouldn't know the answer to this, but maybe the CI environment is just weird and this won't happen in production? What happens if you run it locally? Does it work then?

We enabled huge pages on our production VMs (on GCE) that the FuzzBench service runs the experiments on. The CI runs on a completely different infrastructure (on GitHub).

File: .github/workflows/fuzzers.yml (hunk 1 above)
Comment (jonathanmetzman) [reply .github/workflows/fuzzers.yml:39]: Let's do one for now and if it does better we can try it on the variant.

Comment (jonathanmetzman) [reply .github/workflows/fuzzers.yml:39]: If you think we're likely to miss something by this approach (I don't know much about the big page tables and know nothing about fastcgs_lm) then we can keep. I'm just trying to cut back on unnecessary runs by multiplying features.

**Comment by alifahmed** [comment]
> > @lszekeres @jonathanmetzman I rerun the CI, but shmget() still failing to access huge pages. I am not sure where the problem is. Any idea what went wrong?
> 
> I personally wouldn't know the answer to this, but maybe the CI environment is just weird and this won't happen in production? What happens if you run it locally? Does it work then?

@jonathanmetzman @lszekeres I tried to run locally and have mixed experience. If I enable huge pages outside docker and then run, it works perfectly:
```
# echo 8 > /proc/sys/vm/nr_hugepages
$ make run-fastcgs_lm-<bench>
```

On the other hand, it does not work in the following case:
```
# echo 0 > /proc/sys/vm/nr_hugepages
$ make run-fastcgs_lm-<bench>
```

I thought huge page is being enabled by the docker instance, and both cases should work.

**Comment by jonathanmetzman** [comment]
I guess if it fails in CI than could you please remove it from CI?
Thanks!

**Comment by alifahmed** [comment]
> 
> 
> I guess if it fails in CI than could you please remove it from CI?
> Thanks!

Done.

File: .github/workflows/fuzzers.yml (hunk 1 above)
Comment (alifahmed) [reply .github/workflows/fuzzers.yml]: Done. Kept fastcgs_lm_huge and removed fastcgs_huge.

**Pull request merged**